model/src/pyrenew/convolve.py

```diff
@@ -19,68 +21,146 @@
-        two stages of convolution.
+    arrays_to_convolve : tuple[ArrayLike, ArrayLike]
+        A tuple of two 1D jax arrays, one for
+        each of the two stages of convolution.
+        The first entry in the arrays_to_convolve
+        tuple will be convolved with the
+        current history subset array first, the
+        the second entry will be convolved with
+        it second.
     transforms : tuple[Callable, Callable]
-        A tuple of two functions, each transforming the output of the dot product at each
-        convolution stage.
+        A tuple of two functions, each transforming the
+        output of the dot product at each
+        convolution stage. The first entry in the transforms
+        tuple will be applied first, then the second will
+        be applied.
 
     Returns
     -------
     Callable
-        A scanner function that applies two sequential convolutions and transformations.
-        It takes a history subset and a tuple of multipliers, computes the transformations
-        and dot products, and returns the updated history subset and a tuple of new values.
+        A scanner function that applies two sets of
+        convolution, multiply, and transform operations
+        in sequence to construct a new array by scanning
+        along a pair of input arrays that are equal in
+        length to each other.
+
+    Notes
+    -----
+    Using the same notation as in the documentation for
+    :func:`new_convolve_scanner`, this function aids in
+    applying the iterative operation:
+
+    .. math::
+        \begin{aligned}
+        Y(t) &= f_1 \left(m_1(t)
+           \begin{bmatrix}
+                X(t - n) \\
+                X(t - n + 1) \\
+                \vdots{} \\
+                X(t - 1)
+        \end{bmatrix} \cdot{} \mathbf{d}_1 \right) \\ \\
+        X(t) &= f_2 \left(
+           m_2(t) Y(t)
+        \begin{bmatrix} X(t - n) \\ X(t - n + 1) \\
+        \vdots{} \\ X(t - 1)\end{bmatrix} \cdot{} \mathbf{d}_2 \right)
+        \end{aligned}
+
+    Where :math:`\mathbf{d}_1` and :math:`\mathbf{d}_2` are vectors of
+    length :math:`n`, :math:`m_1(t)` and :math:`m_2(t)` are scalars
+    for each value of time :math:`t`, and :math:`f_1` and :math:`f_2`
+    are scalar-valued functions.
     """
-    d1, d2 = dists
+    arr1, arr2 = arrays_to_convolve
     t1, t2 = transforms
 
     def _new_scanner(
         history_subset: ArrayLike,
         multipliers: tuple[float, float],
     ) -> tuple[ArrayLike, tuple[float, float]]:  # numpydoc ignore=GL08
         m1, m2 = multipliers
-        m_net1 = t1(m1 * jnp.dot(d1, history_subset))
-        new_val = t2(m2 * m_net1 * jnp.dot(d2, history_subset))
+        m_net1 = t1(m1 * jnp.dot(arr1, history_subset))
+        new_val = t2(m2 * m_net1 * jnp.dot(arr2, history_subset))
         latest = jnp.hstack([history_subset[1:], new_val])
         return latest, (new_val, m_net1)
```

Comment:
@gvegayon Do you understand why `m_net1` is returned?

Reply:
1. It's the result of the first dot/multiply/transform operation, so it's useful for tracking and debugging in many cases.

2.In the specific case of a renewal with infection feedback, `m_net1` is the adjusted (with feedback) $\mathcal{R}(t)$ value. 

Should be better documented.

Reply:
How does that work with what `jax.lax.scan` expects?

> `f` returns a pair where the first element represents a new value for the loop carry and the second represents a slice of the output.

Reply:
This bit from the same docs is the key:

> Unlike that Python version, both `xs` and `ys` may be arbitrary pytree values, and so multiple arrays can be scanned over at once and produce multiple output arrays. `None` is actually a special case of this, as it represents an empty pytree.

In this case, we end up producing a tuple of arrays, the first an array of `new_val` values, the second an array of `m_net1` values.